blaze-core/src/main/scala/org/http4s/blaze/util/ProcessWriter.scala

```diff
@@ -52,7 +53,7 @@ trait ProcessWriter {
   final private def go(p: Process[Task, ByteVector], stack: List[StackElem], cb: CBType): Unit = p match {
     case Emit(seq) if seq.isEmpty =>
       if (stack.isEmpty) writeEnd(ByteVector.empty).onComplete(completionListener(_, cb))
-      else go(Try(stack.head.apply(End).run), stack.tail, cb)
+      else Trampoline(go(Try(stack.head.apply(End).run), stack.tail, cb))
```

Comment:
I agree with you, but my terrible habit of prematurely optimizing made me do it this way. :smile: With all this nonsense, the common case of a single `Emit` will not need to go through the trampoline and avoid creating a `Queue` etc. I actually wrote it the way you suggested first but my bad habit got the best of me.